alpaca/data/models/auctions.py

```diff
@@ -0,0 +1,108 @@
+        condition (str): The condition of the auction.
+        price (float): The price of the auction.
+        size (float): The size of the auction.
+        exchange (str): The exchange of the auction.
+        auction_type (str): The type of auction. OPEN or CLOSE.
+
+    """
+
+    symbol: str
+    timestamp: datetime
+    condition: str
+    price: float
+    size: float
+    exchange: str
+    auction_type: str
+
+    def __init__(self, symbol: str, raw_data: RawData) -> None:
+        """Instantiates an auction
+
+        Args:
+            raw_data (RawData): Raw unparsed auction data from API, contains ohlc and other fields.
+        """
+        mapped_auction = {}
+
+        if raw_data is not None:
+            mapped_auction = {
+                AUCTION_MAPPING[key]: val
+                for key, val in raw_data.items()
+                if key in AUCTION_MAPPING
+            }
+
+        super().__init__(symbol=symbol, **mapped_auction)
+
+
+class AuctionSet(BaseDataSet, TimeSeriesMixin):
+    """A collection of Auctions.
+
+    Attributes:
+        data (Dict[str, List[Auction]]): The collection of Auctions keyed by symbol.
+    """
+
+    data: Dict[str, List[Auction]] = {}
+
+    def __init__(self, raw_data: RawData, sort: Sort) -> None:
+        """A collection of Auctions.
+
+        Args:
+            raw_data (RawData): The collection of raw auction data from API keyed by Symbol.
+            sort (Sort): The sort order of the auctions.
+        """
+        parsed_auctions = {}
+
+        raw_auctions = raw_data
+
+        if raw_auctions is not None:
+            for symbol, auctions in raw_auctions.items():
+
+                auction_data = []
+                for auction in auctions:
+                    c = auction.get("c")
+                    o = auction.get("o")
+
+                    if sort == Sort.DESC:
+                        if c is not None:
+                            for close_auction in c:
+                                if close_auction:
+                                    close_auction["at"] = "CLOSE"
+                            auction_data.extend(c)
+                        if o is not None:
+                            for open_auction in o:
+                                if open_auction:
+                                    open_auction["at"] = "OPEN"
+                            auction_data.extend(o)
+                    else:  # NOTE: None and ASC are the same
+                        if o is not None:
+                            for open_auction in o:
+                                if open_auction:
+                                    open_auction["at"] = "OPEN"
+                            auction_data.extend(o)
+                        if c is not None:
+                            for close_auction in c:
+                                if close_auction:
+                                    close_auction["at"] = "CLOSE"
+                            auction_data.extend(c)
```

Comment:
How about below logic?
```suggestion
                    c = auction.get("c")
                    if c is None:
                        c = []
                    o = auction.get("o")
                    if o is None:
                        o = []

                    for close_auction in c:
                        close_auction["at"] = "CLOSE"
                    for open_auction in o:
                        open_auction["at"] = "OPEN"

                    if sort == Sort.DESC:
                        auction_data.extend(c)
                        auction_data.extend(o)
                    else:  # NOTE: None and ASC are the same
                        auction_data.extend(o)
                        auction_data.extend(c)
```

Reply:
> .get

I think both are fine. `c = [] if c is None else c` or .get("c", [])

Reply:
Thanks. Yeah, somehow, `.get("c", [])` does not work for me. checking with below snippets.
```
from datetime import datetime
from zoneinfo import ZoneInfo

from alpaca.common.enums import Sort
from alpaca.data.historical.stock import StockAuctionsRequest, StockHistoricalDataClient


api_key="API KEY"
secret_key="SECRET"
client = StockHistoricalDataClient(api_key, secret_key)

req = StockAuctionsRequest(
    symbol_or_symbols=["SPY"],
    start=datetime(2025, 4,2),
    end=datetime(2025, 4, 9, tzinfo=ZoneInfo("America/New_York")),
    sort=Sort.DESC,
    )
ret = client.get_stock_auctions(req)

```